### PR TITLE
charts: fix several bugs, make the default config work

### PR DESCRIPTION
**Pull request by iawia002**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

After this patch, installing the chart using the default values ​​will succeed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

/cc @warjiang

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,8 +23,6 @@ pnpm-debug.log*
 *.sln
 *.sw?
 _output
-# sub chart tgz
-charts/*/charts
```
Comment (iawia002) [inline .gitignore:27]: This also needs to be committed to the repository, making it convenient for users to install directly.

File: charts/karmada-dashboard/values.yaml [hunk 2]
```diff
@@ -19,20 +19,14 @@ global:
...
 ## Defaults to 'IfNotPresent', you can set to 'Always' in cases of special needs
 karmadaDashboardImagePullPolicy: &karmadaDashboardImagePullPolicy IfNotPresent
 
 podDisruptionBudget: &podDisruptionBudget { }
 
-## @param installMode "host" and "agent" are provided
-## "host" means install karmada in the control-cluster
-## "agent" means install agent client in the member cluster
-## "component" means install selected components in the control-cluster
-installMode: "host"
```
Comment (iawia002) [inline charts/karmada-dashboard/values.yaml:34]: This is completely unnecessary; the dashboard currently doesn't require this kind of config.

**Comment by warjiang** [comment]
I'll check it later today.

File: charts/karmada-dashboard/values.yaml [hunk 3]
```diff
@@ -100,7 +94,8 @@ api:
       maxUnavailable: 0
       maxSurge: 50%
   ## @param apikubeconfig kubeconfig of the api
```
Comment (warjiang) [inline charts/karmada-dashboard/values.yaml:96]: could you help to update the comment for `api.kubeconfigName` and `api.kubeconfigContext`

File: charts/karmada-dashboard/values.yaml [hunk 4]
```diff
@@ -179,4 +174,4 @@ web:
   # runAsUser: 1000
   service:
     type: ClusterIP
-    port: 8000
\ No newline at end of file
+    port: 8000
```
Comment (warjiang) [inline charts/karmada-dashboard/values.yaml:177]: 👍

**Comment by warjiang** [comment]
@iawia002 I've checked it locally, it works 🚀 . Some comments need you to recheck them ~

File: .gitignore (hunk 1 above)
Comment (iawia002) [reply .gitignore:27]: I remember that Karmada had submitted this to the repo before, but it seems I was mistaken.

**Comment by iawia002** [comment]
All comments addressed, PTAL

**Pull request merged**